typescript/packages/plugins/renzo/src/renzo.service.ts

```diff
@@ -0,0 +1,254 @@
+                abi: RENZO_ABI,
+                functionName: "depositToken",
+            });
+
+            console.log(`[Renzo] Deposit token type:`, typeof depositToken);
+            console.log(
+                `[Renzo] Deposit token value:`,
+                JSON.stringify(depositToken)
+            );
+
+            const depositTokenAddress = (
+                depositToken as { value: `0x${string}` }
+            ).value;
+
+            console.log(
+                `[Renzo] Contract deposit token: ${depositTokenAddress}`
+            );
+
+            if (
+                parameters._token.toLowerCase() !==
+                depositTokenAddress.toLowerCase() // Now we can safely call toLowerCase()
+            ) {
+                console.error(`[Renzo] Token mismatch:
+                    Provided: ${parameters._token}
+                    Expected: ${depositTokenAddress}`);
+                throw new Error(
+                    `Invalid token: ${parameters._token}. Expected deposit token: ${depositTokenAddress}`
+                );
+            }
+            console.log(`[Renzo] Token validation successful`);
+
+            console.log(`[Renzo] Initiating deposit transaction with:
+                Amount In: ${parameters._amountIn}
+                Min Out: ${parameters._minOut}
+                Deadline: ${parameters._deadline}`);
+
+            const deadline = BigInt(Math.floor(Date.now() / 1000) + 300); // current time + 5 minutes
+
+            const hash = await walletClient.sendTransaction({
+                to: renzoDepositAddress,
+                abi: RENZO_ABI,
+                functionName: "deposit",
+                args: [
+                    parameters._amountIn,
+                    parameters._minOut,
+                    deadline, // Use the calculated deadline
+                ],
+            });
+            console.log(
+                `[Renzo] Deposit transaction successful. Hash: ${hash.hash}`
+            );
+
+            return hash.hash;
+        } catch (error) {
+            console.error(`[Renzo] Deposit failed with error:`, error);
+            throw Error(`Failed to deposit ERC20: ${error}`);
+        }
+    }
+
+    @Tool({
+        name: "renzo_deposit_eth",
+        description: "Deposit ETH into Renzo",
+    })
+    async depositETH(
+        walletClient: EVMWalletClient,
+        parameters: DepositETHParams
+    ): Promise<string> {
+        try {
+            console.log(
+                `[Renzo] Starting ETH deposit with parameters:`,
+                parameters
+            );
+
+            const { renzoDepositAddress } = getRenzoAddresses(
+                walletClient.getChain().id
+            );
+            console.log(
+                `[Renzo] Using deposit address: ${renzoDepositAddress}`
+            );
+
+            const _minOut = parseUnits(parameters._minOut, 18);
+            const value = parseUnits(parameters._value, 18);
```

Comment:
No need to add this. You can receive it in basis units and there is a tool that already converts into basis units in the erc20 plugin